src/main/kotlin/com/openlattice/directory/UserDirectoryService.kt

```diff
@@ -22,40 +22,78 @@
 package com.openlattice.directory
 
 import com.auth0.json.mgmt.users.User
+import com.codahale.metrics.annotation.Timed
 import com.google.common.collect.ImmutableMap
 import com.hazelcast.core.HazelcastInstance
 import com.hazelcast.core.IMap
 import com.openlattice.auth0.Auth0TokenProvider
+import com.openlattice.authentication.Auth0Configuration
 import com.openlattice.client.RetrofitFactory
 import com.openlattice.datastore.services.Auth0ManagementApi
 import com.openlattice.directory.pojo.Auth0UserBasic
 import com.openlattice.hazelcast.HazelcastMap
 import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
 
 
-open class UserDirectoryService(auth0TokenProvider: Auth0TokenProvider, hazelcastInstance: HazelcastInstance) {
-    companion object {
-        private val logger = LoggerFactory.getLogger(UserDirectoryService::class.java)
-        const val DEFAULT_PAGE_SIZE = 100
+interface UserDirectoryService {
+    @Timed
+    fun getAllUsers(): Map<String, User>
+
+    @Timed
+    fun getUser(userId: String): User
+
+    //TODO: Switch over to a Hazelcast map to relieve pressure from Auth0
+    @Timed
+    fun searchAllUsers(searchQuery: String): Map<String, Auth0UserBasic>
+}
+
+private val logger = LoggerFactory.getLogger(UserDirectoryService::class.java)
+private const val DEFAULT_PAGE_SIZE = 100
+
+@Service
+class LocalUserDirectoryService(auth0Configuration: Auth0Configuration) : UserDirectoryService {
+    val users = auth0Configuration.users.associateBy { it.id }
+    override fun getAllUsers(): Map<String, User> {
+        return users
+    }
+
+    override fun getUser(userId: String): User {
+        return users.getValue(userId)
+    }
+
+    override fun searchAllUsers(searchQuery: String): Map<String, Auth0UserBasic> {
+        return users.values.filter { user ->
+            (listOf(user.email, user.name, user.nickname, user.givenName, user.familyName, user.username) +
+                    user.identities.map { it.userId } + user.identities.map { it.connection })
+                    .any { it.contains(searchQuery) }
```

Comment:
I don't think this will work, bc the query is pre-formatted: 
`String exactEmailSearchQuery = "email.raw:\"" + emailSearchQuery + "\"";`
See in: `PrincipalDirectoryController.searchAllUsersByEmail`

Reply:
I just inverted it. Search doesn't need to be good.